### ISSUE TITLE
[4.0] UuidVX::fromString should fail if provided string belongs to a different version

### ISSUE THREAD

**Issue by jacekkarczmarczyk** [ramsey/uuid#301]
## Describe the bug

Specific version factory, unlike the general one (`RamsayUuid::uuidX()`) should fail if provided string belongs to a different version

## To Reproduce...

```php
        $uuid4String = RamseyUuid::uuid4()->toString();
        $uuid = UuidV1::fromString($uuid4String);
        assert($uuid instanceof UuidV1);
```

## Expected behavior

`UuidV1::fromString` should throw InvalidArgumentException

## Screenshots or output

- `UuidV1::fromString` returns instance of UuidV4
- assert fails

## Environment details

 - OS: Windows 10
 - PHP version: 7.4
 - ramsey/uuid version: 4.0

**Comment by ramsey** [comment]
Good catch. This is a result of these classes extending `Ramsey\Uuid\Uuid` right now. Technically, they shouldn't even have the static methods, since these methods are not part of the interface. In version 5.0, `Ramsey\Uuid\Uuid` will be a final, static class, and these subclasses will not have these static methods, so I'm not sure whether this requires any action at this point.

**Comment by jacekkarczmarczyk** [comment]
Is 5.0 coming soon?

**Comment by ramsey** [comment]
Not until PHP 8. I just released 4.0 two days ago. :-)
